src/main/kotlin/io/arrowkt/service/interpreter/InterestPostingService.kt

```diff
@@ -0,0 +1,41 @@
+package io.arrowkt.service.interpreter
+
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.right
+import arrow.data.EitherT
+import arrow.data.EitherTPartialOf
+import arrow.data.Kleisli
+import arrow.effects.ForIO
+import arrow.effects.IO
+import io.arrowkt.Amount
+import io.arrowkt.model.Account
+import io.arrowkt.service.AccountServiceException
+import io.arrowkt.service.ClosedAccount
+import io.arrowkt.service.InterestPostingService
+import java.math.BigDecimal
+
+object InterestPostingService : InterestPostingService<Account, Amount> {
+
+    override fun calculateInterest(): Kleisli<EitherTPartialOf<ForIO, AccountServiceException>, Account, Amount> =
+        Kleisli { account ->
```

Comment:
Maybe a helper function will make this less noisy.

Reply:
fun <R, A> bla(f: (R) -> A) = Kleisli { dep -> EitherT( IO { f(dep) }) }

Reply:
Tried this... but type inference is failing me :disappointed: